### DISCUSSION TITLE
Find and connect isolated pores

### DISCUSSION THREAD

**Discussion by Sgobro** [PMEAL/OpenPNM#2716] [Q&A] [answered]
Hi OpenPNM team, I'm trying to understand the connectivity of my network created from network_from_porespy; using the function find_connected_pores I find 2777/10172 connected pores but using network.isolated_pores I found just 10 pores, also through check_health I found just 10 pores, and also using find_isolate_clusters I found just 10 pores, where are the other and can I find them? And after that I want to connect them to have a fully connected pore network model.
Thank you so much

**Comment by jgostick** [comment, marked as answer]
If I understand you correctly, you are finding 10 *isolated* pores.  These are among the 2777 total pores.  You need to trim these isolated pores so that the solvers can work.  I actually created a fairly detailed example on this [here](https://openpnm.org/examples/reference/networks/managing_clustered_networks.html)

**Reply by Sgobro** [reply]
Hi no, I have 10172 pores in my network, using Network.find_connected_pores(throats=['all']) I found 2777 pores, now I want to find the unconnected pores and connect them to the network, because I want to obtain a fully connected network.

**Reply by jgostick** [reply]
Ok, so isolated pores are literally solo pores with no neighbors.  There are also 'isolated clusters' which also problematic, but they are groups. The example I linked to also talks about how to deal with that.

**Reply by Sgobro** [reply]
Thanks a lot, I found the way!
